pkg/reconciler/kubernetes/adapter_test.go

```diff
@@ -55,13 +56,16 @@ var expectedResourcesWithNs = []*Resource{
 var expectedLabels = map[string]string{"test-interceptor": "test-label"}
 
 type testInterceptor struct {
-	result InterceptionResult
-	err    error
+	err error
 }
 
-func (i *testInterceptor) Intercept(resource *unstructured.Unstructured, _ string) (InterceptionResult, error) {
-	resource.SetLabels(expectedLabels)
-	return i.result, i.err
+func (i *testInterceptor) Intercept(resources map[string][]*unstructured.Unstructured, _ string) error {
```

Comment:
Maybe for the later PR: could we make the map to an object which offers some convenient looping and getter function.

E.g.

     type Resources map[string][]*unstructured.Unstructured

     func (r resources) Visit(callback func(kind string, u *unstructured.Unstructured) error) error {
     	for kind := range r {
     		for _, resource := range r[kind] {
     			if err := callback(kind, resource); err != nil {
     				return err
     			}
     	     	}
     	}
     	return nil
     }

    func (r resources) Get(kind string) []*unstructured.Unstructured {
      	return r[kind]
     }

Reply:
Done